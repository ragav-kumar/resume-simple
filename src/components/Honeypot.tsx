import React from "react";
import styled from "styled-components";
import { useField } from "formik";

interface HoneypotProps {
	name: string;
}

/**
 * Fairly simple form honeypot to filter out the dumber spammers
 */
export const Honeypot = ( { name }: HoneypotProps ) => {
	const [ field ] = useField<string>(name);
	return (
		<Label>
			<input
				{...field}
				autoComplete="off"
				type="text"
				placeholder="Your name here"
			/>
		</Label>
	);
}
const Label = styled.label`
	opacity: 0;
	position: absolute;
	top: 0;
	left: 0;
	height: 0;
	width: 0;
	z-index: -1;
`;

