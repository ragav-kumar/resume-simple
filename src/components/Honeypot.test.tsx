import React from "react";
import renderer from 'react-test-renderer';
import { Honeypot } from "./Honeypot";

jest.mock('formik', () => ({
	...jest.requireActual<any>('formik'),
	useField: <T extends unknown>( name: string ): [ any, {}, {} ] => [ {
		name,
		value: "dummy",
		onChange: jest.fn,
		onBlur: jest.fn,
		multiple: undefined,
		checked: undefined,
	}, {}, {} ]
}));

describe('<Honeypot /', () => {
	it('should render correctly', () => {
		const tree = renderer
			.create(<Honeypot name="test-honeypot"/>)
			.toJSON();
		expect(tree).toMatchSnapshot();
	});
})


