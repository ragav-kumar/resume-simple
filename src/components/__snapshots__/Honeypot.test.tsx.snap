// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Honeypot / should render correctly 1`] = `
<label
  className="sc-bdfBQB aHUYv"
>
  test-honeypot
  <input
    autoComplete="off"
    name="test-honeypot"
    onBlur={[Function]}
    onChange={[Function]}
    placeholder="Your name here"
    type="text"
    value="dummy"
  />
</label>
`;
